src/main/java/com/rental/rentalApp/controller/ReviewController.java

```diff
@@ -18,35 +14,47 @@
-        this.reviewRepository = reviewRepository;
-    }
-
-    @GetMapping("top")
-    public ResponseEntity<Iterable<Review>> getTopReviews(
-            @RequestParam(required = false) Integer limit) {
-        List<Review> reviews = reviewRepository.findAll();
-        return Optional.of(reviews).<ResponseEntity<Iterable<Review>>>map(reviewList -> ResponseEntity.ok(reviewList.stream().sorted(Comparator.comparing(Review::getRating).reversed()).limit(limit == null ? reviews.size() : limit).toList())).orElseGet(() -> ResponseEntity.badRequest().build());
-    }
-
-    @GetMapping("")
-    public ResponseEntity<Iterable<Review>> getReviews(
-            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime startDateTime,
-            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime endDateTime) {
-
-        List<Review> reviews = reviewRepository.findAll();
-
-        if (startDateTime != null && endDateTime != null)
-            return Optional.of(reviews).<ResponseEntity<Iterable<Review>>>map(reviewList -> ResponseEntity.ok(reviewList.stream().filter(review -> review.getDate().isAfter(startDateTime) && review.getDate().isBefore(endDateTime)).toList())).orElseGet(() -> ResponseEntity.badRequest().build());
-        else if (startDateTime != null)
-            return Optional.of(reviews).<ResponseEntity<Iterable<Review>>>map(reviewList -> ResponseEntity.ok(reviewList.stream().filter(review -> review.getDate().isAfter(startDateTime)).toList())).orElseGet(() -> ResponseEntity.badRequest().build());
-        else if (endDateTime != null)
-            return Optional.of(reviews).<ResponseEntity<Iterable<Review>>>map(reviewList -> ResponseEntity.ok(reviewList.stream().filter(review -> review.getDate().isBefore(endDateTime)).toList())).orElseGet(() -> ResponseEntity.badRequest().build());
-        else
-            return Optional.of(reviews).<ResponseEntity<Iterable<Review>>>map(ResponseEntity::ok).orElseGet(() -> ResponseEntity.badRequest().build());
-
-    }
-
+	private final ReviewRepository reviewRepository;
+
+	public ReviewController(ReviewRepository reviewRepository) {
+		this.reviewRepository = reviewRepository;
+	}
+
+	@GetMapping("top")
+	public ResponseEntity<Iterable<Review>> getTopReviews(
+			@RequestParam(required = false) Integer limit) {
+		List<Review> reviews = reviewRepository.findAll();
+		return Optional.of(reviews).<ResponseEntity<Iterable<Review>>>map(
+				reviewList -> ResponseEntity.ok(reviewList.stream().sorted(Comparator.comparing(Review::getRating).reversed())
+						.limit(limit == null ? reviews.size() : limit).toList()))
+				.orElseGet(() -> ResponseEntity.badRequest().build());
+	}
+
+	@GetMapping("")
+	public ResponseEntity<Iterable<Review>> getReviews(
+			@RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime startDateTime,
```

Comment:
It all looks a lot neater to me. I like it!

I don't know if maybe while we're here, could we change the requested pattern of the dates here to a simpler format, as ISO.DATE_TIME will be a bit cumbersome to use when running queries to showcase our endpoints tomorrow. Maybe changing the name of the variable to a simpler one would help as well. I can alternatively make a separate PR for this change, I just thought that since we're low on time and since it's a small change we could maybe just do it here. What do you think?
```suggestion
			@RequestParam(name = "start", required = false) @DateTimeFormat(pattern = "yyyy-MM-dd") LocalDateTime startDateTime,
```